BankManagerConsoleApp/Model/Bank/Bank.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Bank.swift
+//  BankManagerConsoleApp
+//
+//  Created by 우롱차, 민성 on 2022/04/28.
+//
+
+import Foundation
+
+final class Bank {
+    private var clientQueue: Queue<Client>
+    private let clerkCount: Int
+    private let spendingTimeForAClient: Double
+    private(set) var totalWorkingTime: Double = 0
+    private(set) var finishedClientCount = 0
+
+    private lazy var bankClerkQueue: Queue<BankClerk> = {
+        var bankClerkQueue = Queue<BankClerk>()
+
+        for _ in 1...clerkCount {
+            let bankClerk = BankClerk(bank: self, spendingTimeForAClient: spendingTimeForAClient)
+            bankClerkQueue.enqueue(bankClerk)
+        }
+
+        return bankClerkQueue
+    }()
+
+    init(clientQueue: Queue<Client>, clerkCount: Int, spendingTimeForAClient: Double) {
+        self.clientQueue = clientQueue
+        self.clerkCount = clerkCount
+        self.spendingTimeForAClient = spendingTimeForAClient
+    }
```

Comment:
clerkCount와 spendingTimeForAClient는 상수를 전달받아 Bank에서 프로퍼티로 대입되고 있는데요. Bank 내부에서만 사용되는 상수라면 굳이 전달받지 않고 Bank 내부 자체 상수로 추가해서 사용하면 되지 않을까요?

아래의 제안과 같이 외부 주입은 Bank에서 필요하지만 Bank와 강하게 결합(의존)되어지지 않을 데이터만 전달해주면 좋을 것 같습니다.
```suggestion
    init(clientQueue: Queue<Client>) {
        self.clientQueue = clientQueue
    }
```

Reply:
하지만 이렇게 할 경우 clerkCount나 spendingTimeForAClient가 바꾸는게 굉장히 불편해질텐대 이 경우 저 두개를 바꿔서 테스트하는게 좀 힘들꺼 같아요! 변할일이 없다면 내부에서 상수로 두는게 맞는거 같습니다.